contrib/sddpg/core.c

```diff
@@ -169,12 +170,25 @@ static HB_ERRCODE pgsqlConnect( SQLDDCONNECTION * pConnection, PHB_ITEM pItem )
    PGconn *       pConn;
    ConnStatusType status;
    const char *   pszHost;
+   void **        hbConn;
+   PHB_ITEM       pSecond;
+   int existingConnection = 0;
 
-   pszHost = hb_arrayGetCPtr( pItem, 2 );
-   if( pszHost && ( strncmp( pszHost, "postgresql://", 13 ) == 0 || strchr( pszHost, '=' ) ) )
-      pConn = PQconnectdb( pszHost );
+   pSecond = hb_itemArrayGet( pItem, 2 );
+   if( HB_IS_POINTER( pSecond ) )
```

Comment:
> My overall concern though is that the idea is based on a raw (non-GC-collected, non-typed) 
> pointer to be received, which is unsafe and I also can't see where it can originate from, 
> because there is no facility in Harbour or its contribs that'd return such raw pointer.
> Probably it'd be best to use the GC-pointers inside pgsql, but that means a bunch of
> additional development.

This will need the currently private pointer handling API in `hbpgsql` to be published, 
meaning moving it to a separate `.c` source, maybe renamings, and adding a public `.h` file.
Then making that contrib a dependency of this one and use its API to verify and retrieve the 
pointer.